cms/apps/media/widgets.py

```diff
@@ -0,0 +1,24 @@
+from django import forms
+from django.utils.html import mark_safe
+from django.contrib.admin.widgets import ForeignKeyRawIdWidget
+
+class ImageThumbnailWidget(ForeignKeyRawIdWidget):
+    '''
+    A widget used to display a thumbnail image preview
+    '''
+    def render(self, name, value, attrs=None):
+        # Import here to avoid circular dependencies
+        from .models import File
+        output = []
+
+        if value:
+            file_obj = File.objects.filter(pk=value).first()
+
+            if file_obj:
+                image_url = file_obj.file.url
+                file_name = str(file_obj)
+                output.append(u' <a href="%s" target="_blank"><img src="%s" alt="%s" width="150" height="150"/></a> %s ' % \
```

Comment:
1) There is a cross-site scripting vulnerability here if this is ever used with untrusted data. Always be super extra careful if you ever find yourself using `mark_safe`.

2) No need to u-prefix strings, we dropped Python 2 support in the code cleanup party. You might want to use an f-string here to format the string too.

Reply:
(actually, [`format_html`](https://docs.djangoproject.com/en/2.2/ref/utils/#django.utils.html.format_html) would be better)

Reply:
I've now used format_html instead. Hopefully this implementation is enough to prevent XSS as I understand format_html applies escaping to arguments by default.

Reply:
correct :)